services/QuillLMS/app/models/classroom.rb

```diff
@@ -59,6 +59,12 @@ class Classroom < ApplicationRecord
 
   accepts_nested_attributes_for :classrooms_teachers
 
+  def destroy 
+    classroom_units.each(&:destroy)
+    ClassroomsTeacher.where(classroom_id: self.id).destroy_all
```

Comment:
We need a query here, since this is a foreign key relationship (i.e. not accessible via `record.classrooms_teachers`)